pkgs/development/libraries/hpx/default.nix

```
@@ -1,17 +1,19 @@
-{ lib, stdenv, fetchFromGitHub, boost, cmake, hwloc, gperftools, pkg-config, python3 }:
+{ lib, stdenv, fetchFromGitHub, asio, boost175, cmake, hwloc, gperftools, ninja
+, pkg-config, python3 }:
 
 stdenv.mkDerivation rec {
   pname = "hpx";
-  version = "1.6.0";
+  version = "1.7.1";
 
   src = fetchFromGitHub {
     owner = "STEllAR-GROUP";
     repo = "hpx";
     rev = version;
-    sha256 = "sha256-Fkntfk5AaWtS1x0fXfLSWW/9tvKcCBi1COqgNxurPmk=";
+    sha256 = "1knx7kr8iw4b7nh116ygd00y68y84jjb4fj58jkay7n5qlrxh604";
   };
 
-  buildInputs = [ boost hwloc gperftools ];
+  buildInputs =
+    [ (asio.override { boost = boost175; }) boost175 hwloc gperftools ];
```

```suggestion
  buildInputs = [ asio boost175 hwloc gperftools ];
```

and move the rest to top-level with something like

```
{
  asio = asio.override { boost = boost175; }
  boost = boost175;
}
```

---

Thanks! I've updated this part. Calling the overridden `asio` just `asio` did not work (recursive definition). Did I misunderstand what you were aiming for or is it ok the way I've updated it?